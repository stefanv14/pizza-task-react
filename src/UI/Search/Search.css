.formGroup {
  position: relative;
  width: 30%; }

@media screen and (max-width: 600px) {
  .formGroup {
    width: 80%; } }

.formGroup .inputForm {
  width: 100%; }

.form {
  justify-content: center; }
