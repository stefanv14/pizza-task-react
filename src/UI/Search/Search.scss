.formGroup {
  position: relative;
  width: 30%;
  @media screen and (max-width: 600px) {
    width: 80%;
  }
  .inputForm {
    width: 100%;
  }
}

.form {
  justify-content: center;
}
