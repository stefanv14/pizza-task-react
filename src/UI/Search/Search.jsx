import { faSearch } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import PropTypes from 'prop-types'
import React, { useEffect, useState } from 'react'
import { Form, FormControl, FormGroup } from 'react-bootstrap'
import { useDispatch } from 'react-redux'
import { setSearch } from '../../Redux/Actions/ProductActions/ProductActions'
import './Search.scss'

export const Search = (props) => {
  const { perPage, currency } = props
  const dispatch = useDispatch()

  const [innerWidth, setInnerWidth] = useState(false)
  useEffect(() => {
    if (window.innerWidth < 800) {
      setInnerWidth(true)
    }
  }, [innerWidth])

  const getBySearch = (e) => {
    dispatch(props.dispatch(1, perPage, e, currency))
    dispatch(setSearch(e))
  }

  const style = innerWidth
    ? {
      position: 'absolute',
      top: '.5rem',
      zIndex: 2,
      display: 'block',
      width: '2.375rem',
      height: '1.3rem',
      lineHeight: '1.3rem',
      textAlign: 'center',
      pointerEvents: 'none',
      color: '#aaa',
    }
    : {
      position: 'absolute',
      zIndex: 2,
      display: 'block',
      width: '2.375rem',
      height: '2.375rem',
      lineHeight: '3.375rem',
      textAlign: 'center',
      pointerEvents: 'none',
      color: '#aaa',
      padding: '.4rem',
    }
  return (
    <Form inline className="form d-flex align-items-center mt-3">
      <FormGroup className="has-search formGroup">
        <FormControl
          onChange={(e) => getBySearch(e.target.value)}
          type="text"
          placeholder="Search"
          className="pl-5 inputForm"
        />
        <FontAwesomeIcon icon={faSearch} style={style} />
      </FormGroup>
    </Form>
  )
}
Search.propTypes = {
  perPage: PropTypes.number.isRequired,
  currency: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
}
