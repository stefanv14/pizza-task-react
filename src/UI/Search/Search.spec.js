import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { Search } from './Search'

describe('Search', () => {
  const initialState = {}
  const mockStore = configureStore()
  let store
  test('renders Search component', () => {
    store = mockStore(initialState)
    const { getByText } = render(
      <Provider store={store}>
        <Search />
      </Provider>,
    )
    expect(getByText).toMatchSnapshot()
  })
})
