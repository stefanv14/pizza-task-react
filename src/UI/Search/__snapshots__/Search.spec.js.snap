// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders Search component 1`] = `[Function]`;
